pkg/descheduler/strategies/failedpods.go

```diff
@@ -0,0 +1,178 @@
+package strategies
+
+import (
+	"context"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+// RemoveFailedPods removes Pods that are in failed status phase.
+func RemoveFailedPods(
+	ctx context.Context,
+	client clientset.Interface,
+	strategy api.DeschedulerStrategy,
+	nodes []*v1.Node,
+	podEvictor *evictions.PodEvictor,
+) {
+	if err := validateRemoveFailedPodsParams(strategy.Params); err != nil {
```

Comment:
Do you want to break this out into a `validateAndParse...` params function that returns a struct, similar to in https://github.com/kubernetes-sigs/descheduler/pull/576/files#diff-3ef994f616f4d3726bc08b98eb341fee83b26703bc6179277f5e91286d99b289R51-R60?